helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -1135,6 +1145,34 @@ private void removeMessageFromZK(HelixDataAccessor accessor, Message message,
     }
   }
 
+  private void startMessageRefreshThread() {
```

Comment:
How about let us add a comment about the reason why we add this thread?
Something like:
Async batch read has bug that prevent it from reading all the messages successfully. The issue is separately addressed by pull request [xxxx]. We also add this one as periodical "pull" messages from ZkServer method to preemptively read and process messages from controller.

Reply:
Added this comment in CallbackHandler. Thanks for reminding!